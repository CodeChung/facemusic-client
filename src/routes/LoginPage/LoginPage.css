.LoginPage {
    height: 70%;
    margin: 0 auto;
    width: 40%;
    background: #231F20;
    border-radius: 15px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    padding-bottom: 50px;
    color: whitesmoke;
}

.LoginPage h2 {
    margin: 0;
}

.LoginForm label {
    display: none;
}

.LoginForm input {
    border: none;
    border-bottom: 1px solid gray;
    background-color: #F1FFFA;
    border-radius: 2px;
}

.LoginForm input {
    margin: 15px 0;
    width: 60%;
    font-size: 20px;
}

.LoginForm input:focus {
    outline: none;
}

.LoginForm button {
    width: 30%;
    margin-top: 30px;
    padding: 10px;
    border-radius: 20px;
    border: none;
    background-color: rgb(255, 255, 255);
}

.LoginForm button:focus {
    outline: none;
}

.LoginForm button:active {
    background-color: rgb(163, 163, 163);
}

@media(max-width:768px){
    .LoginPage {
        width: 70%;
    }
}