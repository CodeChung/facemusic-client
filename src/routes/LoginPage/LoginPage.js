import React, { Component } from 'react'
import LoginForm from '../../components/LoginForm/LoginForm'

class LoginPage extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {},
    },
  }

  handleLoginSuccess = () => {
    const { location, history } = this.props
    const destination = (location.state || {}).from || '/'

    history.push(destination)
    console.log(destination)
  }

  render() {
    return (
      <section className='LoginPage'>
        <h2>Login</h2>
        <LoginForm
          onLoginSuccess={this.handleLoginSuccess}
        />
      </section>
    )
  }
}

export default LoginPage