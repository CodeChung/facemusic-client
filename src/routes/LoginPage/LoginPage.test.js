import React from 'react';
import ReactDOM from 'react-dom';
import LoginPage from './LoginPage';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe(`LoginPage Component`, () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<LoginPage />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders an expected LoginPage', () => {
    const wrapper = shallow(<LoginPage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})