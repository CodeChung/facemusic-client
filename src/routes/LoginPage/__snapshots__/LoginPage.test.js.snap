// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage Component renders an expected LoginPage 1`] = `
<section
  className="LoginPage"
>
  <h2>
    Login
  </h2>
  <LoginForm
    onLoginSuccess={[Function]}
  />
</section>
`;
